packages/sdk/src/FlagResolverClient.ts

```diff
@@ -275,6 +297,25 @@ export class CachingFlagResolverClient implements FlagResolverClient {
   }
 }
 
+export function withTelemetryData(
+  fetchImplementation: (request: Request) => Promise<Response>,
```

Comment:
Hmm.. now that we have these two orthogonal fetch modifiers, maybe it would be nicer to pass a FetchBuilder to this function and the other. It's a little nicer than composing them.. wdyt?